File: pp0604a_srednia_arytm/pp0604a.py
t=int(input())
while (t>0):
    line = input().split()
    n = list(map(int, line))
    srednia = sum(n[1:])/n[0]
   
    min = abs(srednia - n[1])
    min_i = 1
    
    for i in range(2,len(n)):
        if abs(srednia-n[i]) < min:
            min_i=i
            min = abs(srednia-n[i])

    print(n[min_i])
    t-=1

File: pp0604a_srednia_arytm/pp0604a.rb
t=gets().to_i

def suma(a)
	sum=0
    a.each {|i| sum += i}
    return sum
end

def srednia_ar(a, n)
	return suma(a)/n.to_f
end

t.times do |i|
    line = gets().split().map(&:to_i)
    n= line[0]
    line.shift()
    srednia = srednia_ar(line, n)
    min_i = 0
    for j in 1...n do
    	if (srednia-line[min_i]).abs>(srednia-line[j]).abs then
    		min_i = j
    	end
    end
    print "#{line[min_i]}\n"
end